File: wlr/OGUR/OGUR/Util/StringStorage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace OGUR.Util
{
    class StringStorage
    {
        private static readonly Dictionary<float, string> s_slots = new Dictionary<float, string>();

        public static string Get(float value)
        {
            if (s_slots.Keys.Count() == 0)
            {
                for (float ii = 0; ii < 1000; ii++)
                {
                    s_slots.Add(ii, ii.ToString());
                }
            }
            if (s_slots.ContainsKey(value))
            {
                return s_slots[value];
            }
            return "**";
        }
    }
}

File: aigilas/src/sps/util/StringStorage.java
package sps.util;

import java.util.HashMap;

public class StringStorage {
    private static final HashMap<Integer, String> __slots = new HashMap<Integer, String>();

    public static String get(float value) {
        if (__slots.keySet().size() == 0) {
            for (int ii = 0; ii < 10000; ii++) {
                __slots.put(ii, Integer.toString(ii));
            }
        }
        if (__slots.containsKey((int) value)) {
            return __slots.get((int) value);
        }
        return Float.toString(value);
    }
}
